### ISSUE TITLE
Combine base `project` with `project-doc`

### ISSUE THREAD

**Issue by samdphillips** [countvajhula/racket-library#1]
Some projects combine the `project` and `project-doc` packages.  It simplifies how many packages need to be managed by the split.

Example:
- https://github.com/rmculpepper/racket-http123

**Comment by countvajhula** [comment]
Interesting - thanks for the suggestion! You know, I wasn't always a fan of the full-blown lib/test/doc breakdown, but I've now come around to the view that it's the most "functional" option - where the library is decomposed into natural elementary components that can be composed in any way we like - typically as the (lib + test + doc) package, but there is flexibility to even do (lib + test + doc + third party plugin + useful development tool + ...).

I can see the appeal of combining docs and tests into the aggregate package while leaving out lib, as we can think about it as "the core library" and "everything else". But it also means that the dependencies aren't as finely distinguished in the package ecosystem as they could be, so that just a mention of `@other-doc[project]` would end up depending on `project` instead of `project-doc` which could in some cases lead to a substantial chain of dependencies that have nothing to do with docs.

In this respect, my gut feeling is that maybe we ought to encourage lib/test/doc use -- in the interest of making the ecosystem of packages leaner and more efficient and more composable -- rather than just for specific libraries to appear simpler.

Wdyt? Of course we always have the option to create an additional template here which uses the twofold breakdown, and I don't mind doing that if you feel strongly about it.

I realize that is a very long-winded response to your simple suggestion, but I agonized over it for a while and this is the result 😂

**Comment by samdphillips** [comment]
I see what you mean.  For me, moving to `packagename` and `packagename-lib` saves the effort of having to register another package.  But for the case of being an example for others to follow having the lib/test/doc split may be useful.

**Comment by countvajhula** [comment]
Yeah, adding 4 packages is a bit of a hassle, especially as you're entering a lot of the same information each time. That's great feedback. I know that many folks have complained about the loose association of these lib/test/doc packages before, and maybe having some kind of formal link via some new ["package concept"](https://docs.racket-lang.org/pkg/Package_Concepts.html) would allow the package registration page to make the UI more convenient. E.g. "(+) Add another package for this project" 🤷
